drivers/sensor/i3g4250d/i3g4250d_spi.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2021 Jonathan Hahn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_i3g4250d
+
+#include <logging/log.h>
+#include "i3g4250d.h"
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+
+#define I3G4250D_SPI_READM      (3 << 6)        /* 0xC0 */
+#define I3G4250D_SPI_WRITEM     (1 << 6)        /* 0x40 */
+
+LOG_MODULE_DECLARE(i3g4250d, CONFIG_SENSOR_LOG_LEVEL);
+
+struct spi_config i3g4250d_spi_conf = {
```

Comment:
Why a special file for handling SPI? Also, the recently introduced `spi_dt_spec` should be used, you can take https://github.com/zephyrproject-rtos/zephyr/pull/35538 as a reviewed example

Reply:
I want to add an i2c counterpart in the future, therefore a special file.

Reply:
`spi_dt_spec` is used now as well.